octodns/provider/route53.py

```diff
@@ -606,28 +1003,44 @@ def get_health_check_id(self, record, ident, geo, create):
             'EnableSNI': healthcheck_protocol == 'HTTPS',
             'FailureThreshold': 6,
             'FullyQualifiedDomainName': healthcheck_host,
-            'IPAddress': first_value,
             'MeasureLatency': healthcheck_latency,
             'Port': healthcheck_port,
             'RequestInterval': 10,
             'ResourcePath': healthcheck_path,
             'Type': healthcheck_protocol,
         }
+        if value:
+            config['IPAddress'] = value
+
         ref = '{}:{}:{}:{}'.format(self.HEALTH_CHECK_VERSION, record._type,
                                    record.fqdn, uuid4().hex[:12])
         resp = self._conn.create_health_check(CallerReference=ref,
                                               HealthCheckConfig=config)
         health_check = resp['HealthCheck']
         id = health_check['Id']
+
+        # Set a Name for the benefit of the UI
+        name = '{}:{} - {}'.format(record.fqdn, record._type,
+                                   value or healthcheck_host)
+        self._conn.change_tags_for_resource(ResourceType='healthcheck',
+                                            ResourceId=id,
+                                            AddTags=[{
+                                                'Key': 'Name',
+                                                'Value': name,
+                                            }])
+        # Manually add it to our cache
+        health_check['Tags'] = {
+            'Name': name
+        }
```

Comment:
This just adds a `Name` tag to the healthchecks that makes them a lot easier to make sense of in the UI...